aten/src/ATen/native/native_functions.yaml

```diff
@@ -170,18 +170,34 @@
     CPU: _assert_async_cpu
     CUDA: _assert_async_cuda
 
+- func: _functional_assert_async(Tensor self, Tensor dep_token) -> Tensor
```

Comment:
I don't think we actually need this. Dynamo always uses the _assert_async.msg variant.